File: app/src/main/java/com/jmdevelopers/workproject/MainActivity.java
package com.jmdevelopers.workproject;

import androidx.viewpager.widget.ViewPager;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;

import com.jmdevelopers.workproject.Fragments.DoarFragment;
import com.jmdevelopers.workproject.Fragments.MinhasDoacoesFragment;
import com.ogaclejapan.smarttablayout.SmartTabLayout;
import com.ogaclejapan.smarttablayout.utils.v4.FragmentPagerItemAdapter;
import com.ogaclejapan.smarttablayout.utils.v4.FragmentPagerItems;

public class MainActivity extends AppCompatActivity {
    private ViewPager viewPager;
    private SmartTabLayout  smartTabLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        viewPager = findViewById(R.id.viewpager);
        smartTabLayout = findViewById(R.id.viewpagertab);
        getSupportActionBar().setElevation(0);
        // configurar action bar

        getSupportActionBar().setTitle("Doações");
        FragmentPagerItemAdapter adapter=new FragmentPagerItemAdapter(getSupportFragmentManager(),
                FragmentPagerItems.with(this).add("Doar", DoarFragment.class).add("Minhas Doações", MinhasDoacoesFragment.class)
                        .create());
        viewPager.setAdapter(adapter);
        smartTabLayout.setViewPager(viewPager);
    }
}

File: app/src/main/java/com/br/vobis/MainActivity.kt
package com.br.vobis

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager
import com.br.vobis.fragments.DonationNeedsFragment
import com.br.vobis.fragments.DonationsFragment
import com.br.vobis.fragments.MyDonationsFragment
import com.google.firebase.auth.FirebaseAuth
import com.ogaclejapan.smarttablayout.SmartTabLayout
import com.ogaclejapan.smarttablayout.utils.v4.FragmentPagerItemAdapter
import com.ogaclejapan.smarttablayout.utils.v4.FragmentPagerItems

class MainActivity : AppCompatActivity() {
    lateinit var mAuth: FirebaseAuth
    private var viewPager: ViewPager? = null
    private var smartTabLayout: SmartTabLayout? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mAuth = FirebaseAuth.getInstance()

        viewPager = findViewById(R.id.viewpage)
        smartTabLayout = findViewById(R.id.viewpagertab)
        supportActionBar!!.elevation = 0f
        // configurar action bar

        supportActionBar!!.title = "Doações"
        val adapter = FragmentPagerItemAdapter(supportFragmentManager,
                FragmentPagerItems.with(this)
                        .add("Doar", DonationsFragment::class.java)
                        .add("Minhas Doações", MyDonationsFragment::class.java)
                        .create())
        viewPager!!.adapter = adapter
        smartTabLayout!!.setViewPager(viewPager)

    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

        val fragment = supportFragmentManager.fragments

        fragment.forEach {
            it.onActivityResult(requestCode, resultCode, data)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        val inflater = menuInflater

        inflater.inflate(R.menu.menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {

        when (item?.itemId) {
            R.id.logout -> {
                mAuth.signOut()
                val intent = Intent(this, PhoneAuthActivity::class.java)

                startActivity(intent)
                finish()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }
}
